packages/admin-next/dashboard/src/routes/products/common/upload-media-form-item.tsx

```diff
@@ -0,0 +1,87 @@
+import { useTranslation } from "react-i18next"
+import { FileType, FileUpload } from "../../../components/common/file-upload"
+import { Form } from "../../../components/common/form"
+
+const SUPPORTED_FORMATS = [
+  "image/jpeg",
+  "image/png",
+  "image/gif",
+  "image/webp",
+  "image/heic",
+  "image/svg+xml",
+]
+
+const SUPPORTED_FORMATS_FILE_EXTENSIONS = [
+  ".jpeg",
+  ".png",
+  ".gif",
+  ".webp",
+  ".heic",
+  ".svg",
+]
+
+export const UploadMediaFormItem = ({
+  form,
+  append,
+}: {
+  form: any
+  append: any
+}) => {
+  const { t } = useTranslation()
+
+  const hasInvalidFiles = (fileList: FileType[]) => {
+    const invalidFile = fileList.find(
+      (f) => !SUPPORTED_FORMATS.includes(f.file.type)
+    )
+
+    if (invalidFile) {
+      form.setError("media", {
+        type: "invalid_file",
+        message: t("products.media.invalidFileType", {
+          name: invalidFile.file.name,
+          types: SUPPORTED_FORMATS_FILE_EXTENSIONS.join(", "),
+        }),
+      })
+
+      return true
+    }
+
+    return false
+  }
+
+  return (
+    <Form.Field
+      control={form.control}
+      name="media"
+      render={() => {
+        return (
+          <Form.Item>
+            <div className="flex flex-col gap-y-4">
+              <div className="flex flex-col gap-y-1">
+                <Form.Label optional>{t("products.media.label")}</Form.Label>
+                <Form.Hint>{t("products.media.editHint")}</Form.Hint>
+              </div>
+              <Form.Control>
+                <FileUpload
+                  label={t("products.media.uploadImagesLabel")}
+                  hint={t("products.media.uploadImagesHint")}
+                  hasError={!!form.formState.errors.media}
+                  formats={SUPPORTED_FORMATS}
+                  onUploaded={(files) => {
+                    form.clearErrors("media")
+                    if (hasInvalidFiles(files)) {
+                      return
+                    }
+
+                    files.forEach((f) => append({ ...f, isThumbnail: false }))
```

Comment:
Q: should `isThumbnail` be provided?

Reply:
You mean by the BE?

Reply:
no here, I mean that here we are always setting it to false, is there a reason, I might miss some context on that

Reply:
Ah ok I see what you mean, this is just temporary, there should be a selector to say if this is a thumbnail or not (we just haven't implemented it yet).

Reply:
oh ok, got it, should we add some todo in those places when we do that? so that if someone jump onto it in between he/she is aware

Reply:
Yup sounds good 👍